core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -171,20 +171,42 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
 
     // if there is any error, may leave user in weird state which can be seen with getUserStatus
     // retrying this call will retry the failures, failures due to already created groups/entries are ok
-    handleFutures(Future.sequence(Seq(
-      toFutureTry(gcsDAO.createProxyGroup(user) flatMap( _ => gcsDAO.addUserToProxyGroup(user))),
-      toFutureTry(for {
+    handleFutures(
+      Future.sequence(Seq(
+        toFutureTry(gcsDAO.createProxyGroup(user) flatMap( _ => gcsDAO.addUserToProxyGroup(user))),
+        toFutureTry(for {
         // these things need to be done in order
-        _ <- dataSource.inTransaction { dataAccess => dataAccess.rawlsUserQuery.save(user) }
-        _ <- dataSource.inTransaction { dataAccess => getOrCreateAllUsersGroup(dataAccess) }
-        _ <- updateGroupMembership(allUsersGroupRef, addUsers = Set(user))
-      } yield ()),
-      toFutureTry(userDirectoryDAO.createUser(user.userSubjectId) flatMap( _ => userDirectoryDAO.enableUser(user.userSubjectId)))
-
-    )))(_ => RequestCompleteWithLocation(StatusCodes.Created, s"/user/${user.userSubjectId.value}"), handleException("Errors creating user")
+          _ <- dataSource.inTransaction { dataAccess => dataAccess.rawlsUserQuery.save(user) }
+          _ <- dataSource.inTransaction { dataAccess => getOrCreateAllUsersGroup(dataAccess) }
+          _ <- updateGroupMembership(allUsersGroupRef, addUsers = Set(user))
+        } yield ()),
+        toFutureTry(userDirectoryDAO.createUser(user.userSubjectId) flatMap( _ => userDirectoryDAO.enableUser(user.userSubjectId)))
+
+      )).flatMap{ _ => Future.sequence(Seq(toFutureTry(turnInvitesIntoRealAccess(user))))})(_ =>
+      RequestCompleteWithLocation(StatusCodes.Created, s"/user/${user.userSubjectId.value}"), handleException("Errors creating user")
     )
   }
 
+  def turnInvitesIntoRealAccess(user: RawlsUser) = {
+    val groupRefs = dataSource.inTransaction { dataAccess =>
```

Comment:
Getting the access group refs can be done in 1 query. This implementation does 1 query to find, 1 each to load access group. It is not a big deal here but we should be in the mindset of doing things in bulk in the database.

Maybe a rule of thumb should be if you use DBIO.sequence, look for a better way.